project/Boilerplate.scala

```diff
@@ -516,6 +547,12 @@ object Boilerplate {
         else
           s"def traverseN[G[_]: Applicative, Z](f: (${`A..N`}) => G[Z])(implicit traverse: Traverse[F], semigroupal: Semigroupal[F]): G[F[Z]] = Semigroupal.traverse$arity($tupleArgs)(f)"
 
+      val flatMap =
+        if (arity == 1)
+          s"def flatMap[Z](f: (${`A..N`}) => F[Z])(implicit flatMap: FlatMap[F]): F[Z] = flatMap.flatMap($tupleArgs)(f)"
+        else
+          s"def flatMapN[Z](f: (${`A..N`}) => F[Z])(implicit flatMap: FlatMap[F]): F[Z] = flatMap.flatMap$arity($tupleArgs)(f)"
```

Comment:
So I haven't been able to wrap my head around whether this technically belongs in `TupleSemigroupalSyntax`. It's certainly the easiest place to put it, and it works, I'm just confused by the "semigroupal" in the name :)

Reply:
> [Semigroupal](https://typelevel.org/cats/api/cats/Semigroupal.html) captures the idea of composing independent effectful values.

Ah, ok that makes sense :)